app/controllers/api/v1/versions_controller.rb

```diff
@@ -4,13 +4,19 @@ class Api::V1::VersionsController < Api::BaseController
   def show
     return unless stale?(@rubygem)
 
-    if @rubygem.public_versions.count.nonzero?
-      respond_to do |format|
-        format.json { render json: @rubygem.public_versions }
-        format.yaml { render yaml: @rubygem.public_versions }
-      end
+    if params[:version]
+      to_render = @rubygem.public_versions.find_by(number: params[:version])
     else
+      to_render = @rubygem.public_versions
```

Comment:
Let's move this to `def index` instead. We shouldn't need a decision like this in one action.